### ISSUE TITLE
E-Collection Quick Update - Replacing 'Service Field' Public Note or Authentication note clears values from other fields

### ISSUE THREAD

**Issue by fionatinto** [ExLibrisGroup/alma-e-collection-quick-update#37]
Hi Josh,

Firstly, sorry I didn't get back to you regarding issues #24, #30 and #31 .  I haven't tested #24 yet, but am going to do that now after I've logged this to you. If there are any problems I'll comment on the ticket to reopen it; if you don't hear back from me further then assume that everything is now working fine on that one and thanks :)

The specific problems reported in #30 and #31 seem to be sorted now, thank you, however I've encountered a further problem with the Service Field Notes fields.  NB - just to avoid confusion - it is the Service Field Public note and Authentication Note fields that I have been testing on here (which you have to scroll further down the list of fields within the app to get to).

The issue I'm having is occurring in both sandbox and live Almas.  When I attempt to 'Replace' on either of these fields (either replacing an existing value or replacing no value i.e. so populating a previously blank field), if there is anything currently in the other field then it gets cleared which I do not want.  

Please see attached screenshots which hopefully explain.  I also hit a further problem with one specific collection where the update app said the job hadn't completed, but when I checked the collection in Alma it looked like it had.   I don't know if that's just an anomaly with that one electronic collection.  

Let me know if you need anything clarified but hopefully the screenshots document should explain.

Thanks,
Fiona

**Comment by fionatinto** [comment]
[CloudApp ecollection quick update issue 37.docx](https://github.com/ExLibrisGroup/alma-e-collection-quick-update/files/8006202/CloudApp.ecollection.quick.update.issue.37.docx)

screenshot file now attached

**Comment by fionatinto** [comment]
Hi Josh,

I tested issue #24.  It is now fixed in that 'Clear' now works successfully on Service Field > Active Until Field and when you use 'Replace' on Service Field > Active Until field it inputs the correct date now.  Thanks :)

However, when editing this field I came across the problem as I have reported here in issue #37 - ‘Replace’ on the Service Field > Active Until field also has the unwanted side effect of also clearing data from the Service Field > Public Note and Authentication Note fields.

2nd screenshot doc attached.

I only tested this on sandbox Alma, but I presume it will be the same on live.

Thanks,
Fiona

[CloudApp ecollection quick update issue 37 doc 2.docx](https://github.com/ExLibrisGroup/alma-e-collection-quick-update/files/8006303/CloudApp.ecollection.quick.update.issue.37.doc.2.docx)
